src/geometry/cgal/cgalutils.cc

```diff
@@ -383,13 +384,15 @@ bool createPolySetFromNefPolyhedron3(const CGAL::Nef_polyhedron_3<K>& N, PolySet
     LOG(message_group::Error, "Non-manifold mesh created: %1$d unconnected edges", unconnected2);
   }
 
-  ps.reserve(allTriangles.size());
-  for (const auto& t : allTriangles) {
-    ps.append_poly(3);
-    ps.append_vertex(verts[t[0]]);
-    ps.append_vertex(verts[t[1]]);
-    ps.append_vertex(verts[t[2]]);
+  PolySetBuilder builder(verts.size(),allTriangles.size());
+  for(int i=0;i<verts.size();i++)
+    builder.vertexIndex({verts[i][0],verts[i][1],verts[i][2]});
+
+  for(int i=0;i<allTriangles.size();i++)  {
+	const auto &tri=allTriangles[i];
+	builder.appendPoly({tri[0],tri[1],tri[2]});
```

Comment:
Builder may remove degenerates, so the index here may not be correct?
You need to store the indices in the loop above, e.g. `indMap[i] = builder.vertIndex ...`, and do `indMap[tri[0]]`...